### ISSUE TITLE
provide support for the prometheus operator ServiceMonitor CR

### ISSUE THREAD

**Issue by jurgenweber** [banzaicloud/bank-vaults#383]
**Is your feature request related to a problem? Please describe.**
the CoreOS Prometheus Operator is here, offically support it by providing a ServiceMonitor resource in vault-operator

**Describe the solution you'd like**
a flag to choose between the scrape annotations for prometheus vs a service monitor.

**Additional context**
You will need two variants, vault 1.0 and vault 1.1.. an example ServiceMonitor resource below:

```
---
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  labels:
    app: vault
    release: prometheus-operator
  name: prometheus-operator-vault
  namespace: secrets
spec:
  endpoints:
   # this scrapes the statsd exporter in 1.0
    - interval: 30s
      port: prometheus
  # this scrapes the native endpoint in 1.1
    - bearerTokenFile: /etc/prometheus/secrets/vault-prometheus-endpoint-token/vault.token
      interval: 30s
      params:
        format: ['prometheus']
      path: /v1/sys/metrics
      port: api-port
      scheme: https
      tlsConfig:
        caFile: /etc/prometheus/secrets/vault-tls/ca.crt
        certFile: /etc/prometheus/secrets/vault-tls/server.crt
        keyFile: /etc/prometheus/secrets/vault-tls/server.key
        insecureSkipVerify: true
  selector:
    matchLabels:
      app: vault
      vault_cr: vault
```

**Comment by matyix** [comment]
This is already supported by the Helm chart - see https://github.com/banzaicloud/banzai-charts/blob/master/vault/templates/prometheus-monitorservice.yaml, will add it to the operator as well.

**Comment by matyix** [comment]
Fixed by #386

**Comment by bonifaido** [comment]
Some notes:

The `ServiceMonitor` for a Vault CR will be installed only if `serviceMonitorEnabled: true` is given in the resource manifest, see: https://github.com/banzaicloud/bank-vaults/blob/master/operator/deploy/cr-prometheus.yaml#L49

To make this working together with Prometheus the `coreos/prometheus-operator` has to be installed which registers the `ServiceMonitor` CRD (that is why it is not enabled by default, otherwise the vault-operator would fail reconciling the Vault CR).

The linked example Vault CR above configures a very limited Vault policy called `metrics` which is capable of only `HTTP GET`-ing the `/v1/sys/metrics`. The provide a Vault token with this policy for Prometheus is currently the job of the user (there are various tools for that, like the vault-agent), so a token has to be provided to Prometheus to be able to authenticate against this endpoint, how to do that properly/conveniently in an automated way is a matter of discussion.
